algorithms/src/Kokkos_Sort.hpp

```diff
@@ -624,6 +632,18 @@ std::enable_if_t<Kokkos::is_execution_space<ExecutionSpace>::value> sort(
   bin_sort.sort(exec, view);
 }
 
+#if defined(KOKKOS_ENABLE_CUDA)
+template <class ViewType>
+std::enable_if_t<
+    Experimental::Impl::has_random_access_iterator<ViewType>::value>
+sort(Cuda const& space, ViewType const& view, bool const = false) {
```

Comment:
if we know that this should accept a kokkos view, I would suggest doing this: 

```
template <class DataType, class ...Properties>
void sort(Cuda const& space, const Kokkos::View<DataType, Properties...> view, bool const = false)
```

Reply:
I was referring to explicitly use Kokkos::View as an argument rather than using `ViewType`. Makes sense? 
(I personally agree on the defaulted arg being iffy)

Reply:
My bad.  Agreed.

Reply:
1. As I already wrote, in this case, I treat SFINAE as a possibility of having a fallback function. Also, I follow Daniel's suggestion of `static_assert` vs `enable_if` ([here](https://github.com/kokkos/kokkos/pull/5183#discussion_r917951595)). But I don't have a strong opinion on this. If you guys find the usage of `static_assert` (hidden inside `begin()` and `end()`) more appropriate, I'm okay with removing `enable_if`.

2. And about the function's signature:
* `bool const = false` - yeah, I don't like it either, but the original function has this default boolean parameter, so I could either go with an unnamed default value or two overloads, one with and the other without bool par.
* `ViewType const& view` vs `const Kokkos::View<DataType, Properties...> view` - original function takes a `ViewType const& view` argument, so I followed that. If `Kokkos::View` is more suitable, I'll change that.

Reply:
That makes sense to me if we already had a fallback impl, but there isn't one

Reply:
I removed SFINAE from CUDA sort impl.

Reply:
The fallback is the current `BinSort` implementation and SFINAE is a safer approach.

Assuming someone was sorting a view with a custom layout, he would now get a compile-error instead of keep doing the same thing.